src/elements/content-sharing/ContentSharing.js

```diff
@@ -110,30 +145,122 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
         }
     }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
 
-    if (errorMessage) {
-        return <ErrorMask errorHeader={<FormattedMessage {...errorMessage} />} />;
+    // Close a notification
+    const handleNotificationClose = React.useCallback(
```

Comment:
This component seems to be doing a lot of things now. It may be worthwhile to break it apart to reduce the number of effects/callbacks in a single component lifecycle.